tools/internal_ci/linux/grpc_xds_resource_cleanup.sh

```diff
@@ -13,61 +13,124 @@
+#######################################
+# The PSM_LB cluster is used by k8s_lb tests.
+# The keep hours is reduced to 6.
+#######################################
+cleanup::job::cleanup_cluster_lb_primary() {
+  cleanup::activate_cluster GKE_CLUSTER_PSM_LB
+  cleanup::run_clean "$1" --mode=k8s --keep_hours=6
+}
 
-# The BASIC cluster is used by url-map tests. Only cleaning the GKE client
-# namespaces, which won't provide much value in debugging. The keep hours is
-# reduced to 6.
-activate_gke_cluster GKE_CLUSTER_PSM_BASIC
-# Invoking the get-crednetials directly, because the
-# gcloud_get_cluster_credentials re-sets readonly Bash variables, which is nice
-# safety mechanism to keep.
-gcloud container clusters get-credentials "${GKE_CLUSTER_NAME}" --zone "${GKE_CLUSTER_ZONE}"
-TARGET_KUBE_CONTEXT="$(kubectl config current-context)"
-python3 -m bin.cleanup.namespace \
-    --project=grpc-testing \
-    --network=default-vpc \
-    --keep_hours=6 \
-    --kube_context="${TARGET_KUBE_CONTEXT}" \
-    --gcp_service_account=xds-k8s-interop-tests@grpc-testing.iam.gserviceaccount.com \
-    --resource_prefix='required-but-does-not-matter' \
-    --td_bootstrap_image='required-but-does-not-matter' --server_image='required-but-does-not-matter' --client_image='required-but-does-not-matter'
+#######################################
+# Secondary PSM_LB cluster is used by k8s_lb tests.
+# The keep hours is reduced to 6.
+#######################################
+cleanup::job::cleanup_cluster_lb_secondary() {
+  cleanup::activate_secondary_cluster_as_primary GKE_CLUSTER_PSM_LB
+  cleanup::run_clean "$1" --mode=k8s --secondary --keep_hours=6
+}
+
+#######################################
+# The BASIC cluster is used by url-map tests. Only cleaning the xds client
+# namespaces; the xds server namespaces are shared.
+# The keep hours is reduced to 6.
+#######################################
+cleanup::job::cleanup_cluster_url_map() {
+  cleanup::activate_cluster GKE_CLUSTER_PSM_BASIC
+  cleanup::run_clean "$1" --mode=k8s --keep_hours=6
+}
 
-# The PSM_LB cluster is used by k8s_lb tests. Only cleaning the GKE client
-# namespaces, which won't provide much value in debugging. The keep hours is
-# reduced to 6.
-activate_gke_cluster GKE_CLUSTER_PSM_LB
-gcloud container clusters get-credentials "${GKE_CLUSTER_NAME}" --zone "${GKE_CLUSTER_ZONE}"
-TARGET_KUBE_CONTEXT="$(kubectl config current-context)"
-python3 -m bin.cleanup.namespace \
+#######################################
+# The SECURITY cluster is used by the security and authz test suites.
+#######################################
+cleanup::job::cleanup_cluster_security() {
+  cleanup::activate_cluster GKE_CLUSTER_PSM_SECURITY
+  cleanup::run_clean "$1" --mode=k8s --keep_hours=6
+}
+
+#######################################
+# Set common variables for the cleanup script.
+#######################################
+cleanup::run_clean() {
+  local job_name="${1:?Usage: cleanup::run_clean job_name}"
```

Comment:
Can we add comment to explain what `job_name` we accept here?

Reply:
Ahahah good point. I tried to figure out what it's accepting and realized I completely forgot myself.